Stats for registered and approved classes

willgearty opened the pull request:
This removes cancelled and rejected classes from the calculations of the class-hours and class-student-hours in the admin dashboard and teacher big board. I think this makes more sense when setting a goal for a program and comparing across multiple programs.

benjaminjkraft commented:
Huh, not sure.  It could be the `sections__` is doing something funky.  You'd have to post a bit more context for the graph for me to tell.  In any case, this looks fine if you can figure out the bug!

willgearty commented:
Alright, I'll look into it. Also, now that I think about it, would it make sense to include both numbers? All hours and non-cancelled hours?

willgearty commented:
That weird bug was related to lunch and is now fixed in #2685. I have an idea for how to better display the registered vs. approved numbers, so let me give that a shot.

willgearty commented:
OK, all set to be reviewed!

willgearty commented on esp/templates/program/modules/bigboardmodule/bigboard.html:
```diff
@@ -24,86 +24,104 @@
+      }
+      {% if right_axis_data %}
+      ,{
+        title: {text: 'class-hours'},
```
Yeah, because we now have two series for each of the right axes. Any ideas for how to keep this general?

willgearty replied:
Never mind, I figured out a better way to do everything.

willgearty commented:
OK, I removed a lot of the duplicated code, generalized the bigboard template a LOT, and added functionality to show a blank graph when there isn't any data (instead of when there was just nothing or a blank spot before).

kkbrum commented on esp/esp/program/modules/handlers/jsondatamodule.py:
```diff
@@ -689,6 +689,16 @@ def message_requests():
+            if cls['subject_duration']:
+                hours["class-hours"] += float(cls['subject_duration'])
+                hours["class-student-hours"] += float(cls['subject_duration']) * (float(cls['class_size_max']) if cls['class_size_max'] else 0)
+                hours["class-registered-hours"] += float(cls['subject_duration']) * float(cls['subject_students']) / float(cls['num_sections'])
```
Why do we divide by the number of sections here?

willgearty replied:
That is a good question...

willgearty replied:
Oh, I just remembered why: `subject_duration` is the sum of the durations of the sections of the class. Each student only attends 1 section, so we need to divide `subject_duration` by `num_sections` to get the actual duration of the class that the student is registered for.

willgearty commented:
OK, it looks like some themes have default line heights, which mess up the display of the numbers on the big board (I've documented this in #2695). I'll work on that in a separate PR.
